src/main/java/edu/byu/cs/canvas/CanvasUtils.java

```diff
@@ -26,47 +27,24 @@ public class CanvasUtils {
      */
     public static Rubric decimalScoreToPoints(Phase phase, Rubric rubric) throws GradingException, DataAccessException {
         RubricConfig rubricConfig = DaoService.getRubricConfigDao().getRubricConfig(phase);
+        EnumMap<Rubric.RubricType, Rubric.RubricItem> rubricItems = new EnumMap<>(Rubric.RubricType.class);
 
-        Rubric.RubricItem convertedPassoffTests = null;
-        Rubric.RubricItem convertedUnitTests = null;
-        Rubric.RubricItem convertedQuality = null;
+        for(Rubric.RubricType type : Rubric.RubricType.values()) {
```

Comment:
It would be helpful to provide an iterating method in `RubricType` that calls a lambda on each of the values with the  `type` and the `rubricItem`. It would simplify this common iterating pattern used in multiple places...

Reply:
That sounds like it would add a bunch of unnecessary code clutter without a ton of added benefit. We would have to add a new functional interface to make that lambda, unless you know of one that already exists and can throw custom exceptions.